DICOM/src/Microsoft.Health.Dicom.Anonymizer.Core/Utility.cs

```diff
@@ -0,0 +1,172 @@
+            EnsureArg.IsNotNull(dateTime, nameof(dateTime));
+
+            Regex dateTimeRegex = new Regex(@"^((?<year>\d{4})(?<month>\d{2})(?<day>\d{2})(?<hour>\d{2})(?<minute>\d{2})(?<second>\d{2})(\.(?<millisecond>\d{1,6}))?(?<timeZone>(?<sign>-|\+)(?<timeZoneHour>\d{2})(?<timeZoneMinute>\d{2}))?)(\s*)");
+            var matches = dateTimeRegex.Matches(dateTime);
+            if (matches.Count != 1)
+            {
+                throw new Exception();
+            }
+
+            var groups = matches[0].Groups;
+
+            int year = groups["year"].Success ? int.Parse(groups["year"].Value) : 0;
+            int month = groups["month"].Success ? int.Parse(groups["month"].Value) : 0;
+            int day = groups["day"].Success ? int.Parse(groups["day"].Value) : 0;
+            int hour = groups["hour"].Success ? int.Parse(groups["hour"].Value) : 0;
+            int minute = groups["minute"].Success ? int.Parse(groups["minute"].Value) : 0;
+            int second = groups["second"].Success ? int.Parse(groups["second"].Value) : 0;
+            int millisecond = groups["millisecond"].Success ? int.Parse(groups["millisecond"].Value) : 0;
+
+            if (groups["timeZone"].Success)
+            {
+                int timeZoneHour = int.Parse(groups["timeZoneHour"].Value) * int.Parse(groups["sign"].Value + "1");
+                int timeZoneMinute = int.Parse(groups["timeZoneMinute"].Value);
+                return new DateTimeObject()
+                {
+                    DateValue = new DateTimeOffset(year, month, day, hour, minute, second, millisecond > 999 ? int.Parse(millisecond.ToString().Substring(0, 3)) : millisecond, new TimeSpan(timeZoneHour, timeZoneMinute, 0)),
+                    HasTimeZone = true,
+                };
+            }
+            else
+            {
+                return new DateTimeObject()
+                {
+                    DateValue = new DateTimeOffset(year, month, day, hour, minute, second, millisecond, default),
+                    HasTimeZone = false,
+                };
+            }
+        }
+
+        public static string GenerateDicomDateString(DateTimeOffset date)
+        {
+            if (date == null)
+            {
+                return null;
+            }
+
+            return date.ToString("yyyyMMdd", CultureInfo.InvariantCulture);
+        }
+
+        public static string GenerateDicomDateTimeString(DateTimeObject date)
+        {
+            if (date == null)
+            {
+                return null;
+            }
+
+            if (date.HasTimeZone == null || !(bool)date.HasTimeZone)
+            {
+                return date.DateValue.ToString("yyyyMMddhhmmss.ffffff", CultureInfo.InvariantCulture);
+            }
+            else
+            {
+                return date.DateValue.ToString("yyyyMMddHHmmss.ffffffzzz", CultureInfo.InvariantCulture).Replace(":", string.Empty);
+            }
+        }
+
+        public static AgeValue ParseAge(string age)
```

Comment:
>ParseAge [](http://example.com/codeflow?start=31&length=8)

so the format of age string is a known format, or you just parse several kinds of them? You can add some comments as example here